LIBFCREPO-457 First pass at a Notification route

cfitz opened the pull request:
Attempts to implements a email route.

https://issues.umd.edu/browse/LIBFCREPO-457

cfitz committed:
LIBFCREPO-457 First pass at a Notification route

Attempts to implements a email route.

https://issues.umd.edu/browse/LIBFCREPO-457

peichman-umd commented on umd-fcrepo-notification/src/main/java/edu/umd/lib/fcrepo/camel/notification/NotificationRouter.java:
```diff
@@ -0,0 +1,31 @@
+	        /**
+	         * Handle fixity events
+	         */
+	        from("{{notification.sender}}")
```
For consistency with the other routes, `notification.sender` should be renamed to `input.stream`.

peichman-umd commented on umd-fcrepo-notification/src/main/java/edu/umd/lib/fcrepo/camel/notification/NotificationRouter.java:
```diff
@@ -0,0 +1,31 @@
+	        from("{{notification.sender}}")
+	        	.log(LoggingLevel.WARN, LOGGER,
+	        		"Sending to ${{notification.recipient}}")
+	                    .to("{{notification.recipient}}");
```
Rather than packing all of the email configuration in the `notification.recipient` property, it would be better to configure things like `notification.recipients` (a list of 1 or more email addresses), `notification.subject` (subject line for the email), and `notification.body` (body of the email message) as separate properties in the config, and set up the routing to email programmatically. Otherwise, we would just add the camel email component to the failure broadcast config.

peichman-umd commented on umd-features/pom.xml:
```diff
@@ -23,7 +23,7 @@
         <groupId>edu.umd.lib.fcrepo</groupId>
         <artifactId>umd-fcrepo-camel-extensions</artifactId>
-        <version>1.3.0-SNAPSHOT</version>
+        <version>1.3.1-SNAPSHOT</version>
```
Version should remain as 1.3.0-SNAPSHOT.

peichman-umd committed:
LIBFCREPO-457. Revert version to 1.3.0-SNAPSHOT.

https://issues.umd.edu/browse/LIBFCREPO-457

peichman-umd committed:
LIBFCREPO-457. Changed notification.sender to input.stream.

https://issues.umd.edu/browse/LIBFCREPO-457

peichman-umd committed:
LIBFCREPO-457. Use separate notification.* properties.

* Use separate notification.recipients and notification.subject properties.
* Attempt to change the message body to be a simple message rather than the full RDF of the fixity result.
* Removed a debugging line from the umd-fcrepo-triplestore route.

https://issues.umd.edu/browse/LIBFCREPO-457

peichman-umd reviewed (approved):
I've addressed the initial review comments in the subsequent commits.

cfitz committed:
LIBFCREPO-457 Update processor to change message body, use SMTP url.

This set the processor output to change the message body to contain a
line pointing to the URI of the failured file. Also uses the SMTP
instead of trying to dynamically build the output.
